lib/spack/spack/compiler.py

```diff
@@ -304,6 +305,12 @@ def __init__(self, cspec, operating_system, target,
             if value is not None:
                 self.flags[flag] = tokenize_flags(value)
 
+        # cache real version value for version checks
+        try:
+            self._real_version = spack.version.Version(self.get_real_version())
```

Comment:
Don't put this here -- make it lazy.  Here we call `real_version` every time a compiler is instantiated.  For a given spack run we really only want to call this for the compilers we're using. 

If you make `real_version` a property and have the property do the caching, you'll only end up shelling out to the compilers you build with.